softlayer/resource_softlayer_scale_group.go

```diff
@@ -389,21 +389,20 @@ func resourceSoftLayerScaleGroupRead(d *schema.ResourceData, meta interface{}) e
 	// the original order in the config and trigger a false change/update.
 	// This should be fine as we don't expect this to change after the group
 	// is created. Else, this code needs to be refactored to use a TypeSet.
-	if vlanTotal == 0 {
+	if _, ok := d.GetOk("network_vlan_ids"); !ok {
 		vlanIds := make([]int, vlanTotal)
 		for i, vlan := range slGroupObj.NetworkVlans {
 			vlanIds[i] = *vlan.NetworkVlanId
 		}
 		d.Set("network_vlan_ids", vlanIds)
 	}
-
 	virtualGuestTemplate := populateMemberTemplateResourceData(*slGroupObj.VirtualGuestMemberTemplate)
 	d.Set("virtual_guest_member_template", virtualGuestTemplate)
 
 	return nil
 }
 
-func populateMemberTemplateResourceData(template datatypes.Virtual_Guest) map[string]interface{} {
+func populateMemberTemplateResourceData(template datatypes.Virtual_Guest) []map[string]interface{} {
```

Comment:
why do you need to return an array of maps? didn't see where you could have more than one map below...